File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
    
    var newList *ListNode
    head := newList
    for list1 != nil && list2 != nil {
        var node *ListNode
        if list1.Val < list2.Val {
            node = list1
            list1 = list1.Next
        } else {
            node = list2
            list2 = list2.Next
        }
        if head == nil {
            head = node
            newList = node
        } else {
            head.Next = node
            head = head.Next
        }
    }
    for list1 != nil {
        var node *ListNode
        node = list1
        list1 = list1.Next
        
        if head == nil {
            head = node
            newList = node
        } else {
            head.Next = node
            head = head.Next
        }
    }
    for list2 != nil {
        var node *ListNode
        node = list2
        list2 = list2.Next
        
        if head == nil {
            head = node
            newList = node
        } else {
            head.Next = node
            head = head.Next
        }
    }
    return newList
}

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode newList = null;
        ListNode head = null;
        while (list1 != null && list2 != null) {
            ListNode node = new ListNode();
            if (list1.val <= list2.val) {
                node.val = list1.val;
                list1 = list1.next;
            } else {
                node.val = list2.val;
                list2 = list2.next;
            }
            if (newList == null) {
                head = node;
                newList = node;
            } else {
                newList.next = node;
                newList = newList.next;
            }
        }
        while (list1 != null) {
            ListNode node = new ListNode();
            node.val = list1.val;
            if (newList == null) {
                head = node;
                newList = node;
            } else {
                newList.next = node;
                newList = newList.next;
            }
            list1 = list1.next;
        }
        while (list2 != null) {
            ListNode node = new ListNode();
            node.val = list2.val;
            if (newList == null) {
                head = node;
                newList = node;
            } else {
                newList.next = node;
                newList = newList.next;
            }
            list2 = list2.next;
        }
        return head;
    }
    
}